lib/puppet/provider/nameservice.rb

```diff
@@ -25,9 +25,12 @@ def initvars
     def instances
       objects = []
       listbyname do |name|
-        objects << new(:name => name, :ensure => :present)
+        canonical_name = name.dup
+        if !name.valid_encoding?
+          name.scrub!
+        end
+        objects << new(:name => name, :ensure => :present, :canonical_name => canonical_name)
```

Comment:
This approach I think is not the one we're going to go with.

in discussion with @Iristyle I think we concluded extending the Etc::Passwd and Etc::Group structs to include fields that hold this "canonical" data in addition to the "scrubbed" data is a cleaner and more complete/correct solution